zone/lua_client.cpp

```diff
@@ -1761,9 +1766,27 @@ void Lua_Client::QuestReward(Lua_Mob target, luabind::adl::object reward) {
 
 	cur = reward["faction"];
 	if (luabind::type(cur) != LUA_TNIL) {
-		try {
-			faction = luabind::object_cast<bool>(cur);
-		} catch (luabind::cast_failed &) {
+		// if it's a table it will be {faction, faction_mod}
+		if (luabind::type(cur) == LUA_TTABLE) {
+			auto item = cur[1];
+			if (luabind::type(item) != LUA_TNIL) {
+				try {
+					quest_reward.faction = luabind::object_cast<uint32>(item);
+				} catch (luabind::cast_failed &) {
```

Comment:
Could pipe these errors to `LogQuests`

Reply:
See @hgtw comment on this code in the previous PR.